modules/auto-update/implementation/ot-auto-updater.js

```diff
@@ -0,0 +1,240 @@
+     * @param {String} remoteVersion - The version of the application in the git repository. 
+     * 
+     * Checks the local version of the application against the remote repository.
+     * @returns {VersionResults} - An object with the results of the version comparison.
+     */
+    async compareVersions() {
+        try {
+            this.logger.info('AutoUpdater - Comparing versions...');
+            const currentVersion = this.readAppVersion();
+            const remoteVersion = await this.readRemoteVersion();
+            this.logger.info(`AutoUpdater - Current version: ${currentVersion}`);
+            this.logger.info(`AutoUpdater - Remote Version: ${remoteVersion}`);
+            if (currentVersion === remoteVersion) {
+                return {
+                    upToDate: true, 
+                    currentVersion
+                };
+            };
+            return {
+                upToDate: false, 
+                currentVersion, 
+                remoteVersion
+            };
+        }catch(e) {
+            this.logger.error(`AutoUpdater - Error comparing local and remote versions. Error message: ${e.message}`);
+            return {
+                upToDate: false, 
+                currentVersion: 'Error', 
+                remoteVersion: 'Error'
+            }
+        }
+    }
+
+    /**
+     * Clones the git repository and installs the update over the local application.
+     * A backup of the application is created before the update is installed.
+     * If configured, a completion command will be executed and the process for the app will be stopped. 
+     * @returns {Boolean} The result of the update.
+     */
+    async forceUpdate() {
+        try {
+            this.logger.info(`AutoUpdater - Updating ot-node from ${REPOSITORY_URL}`);
+            await this.downloadUpdate();
+            await this.backup();
+            await this.installUpdate();
+            await this.installDependencies();
+            this.logger.info('AutoUpdater - Finished installing updated version.');
+            if (this.config.executeOnComplete) await this.promiseBlindExecute(this.config.executeOnComplete);
+            process.exit(1);
+        }catch(e) {
+            this.logger.error(`AutoUpdater - Error updating application. Error message: ${e.message}`);
+            return false;
+        }
+    }
+
+    /**
+     * Copy the files to the app directory, and install new modules
+     * The update is installed from  the configured tempLocation.
+     */
+    async installUpdate() {
+        const source = path.join(this.config.tempLocation, CLONE_SUBDIRECTORY);
+        const destination = appRootPath.path;
+        this.logger.info('AutoUpdater - Installing update...');
+        this.logger.info(`AutoUpdater - Source: ${source}`);
+        this.logger.info(`AutoUpdater - Destination: ${destination}`);
+        await fs.ensureDir(destination);
+        await fs.copy(source, destination);
+        return true;
```

Comment:
It looks like we are not using this return value anywhere so we don't have to return it